js/packages/candy-machine-ui/src/Home.tsx

```diff
@@ -0,0 +1,248 @@
+      !wallet ||
+      !wallet.publicKey ||
+      !wallet.signAllTransactions ||
+      !wallet.signTransaction
+    ) {
+      return;
+    }
+
+    return {
+      publicKey: wallet.publicKey,
+      signAllTransactions: wallet.signAllTransactions,
+      signTransaction: wallet.signTransaction,
+    } as anchor.Wallet;
+  }, [wallet]);
+
+  const refreshCandyMachineState = useCallback(() => {
+    (async () => {
+      if (!anchorWallet) {
+        return;
+      }
+
+      if (props.candyMachineId) {
+        try {
+          const cndy = await getCandyMachineState(
+            anchorWallet,
+            props.candyMachineId,
+            props.connection,
+          );
+          setCandyMachine(cndy);
+        } catch (e) {
+          console.log('There was a problem fetching Candy Machine state');
+          console.log(e);
+        }
+      }
+    })();
+  }, [anchorWallet, props.candyMachineId, props.connection]);
+
+  const onMint = async () => {
+    try {
+      setIsUserMinting(true);
+      document.getElementById('#identity')?.click();
+      if (wallet.connected && candyMachine?.program && wallet.publicKey) {
+        const mintTxId = (
+          await mintOneToken(candyMachine, wallet.publicKey)
+        )[0];
+
+        let status: any = { err: true };
+        if (mintTxId) {
+          status = await awaitTransactionSignatureConfirmation(
+            mintTxId,
+            props.txTimeout,
+            props.connection,
+            true,
+          );
+        }
+
+        if (!status?.err) {
+          setAlertState({
+            open: true,
+            message: 'Congratulations! Mint succeeded!',
+            severity: 'success',
+          });
+        } else {
+          setAlertState({
+            open: true,
+            message: 'Mint failed! Please try again!',
+            severity: 'error',
+          });
+        }
+      }
+    } catch (error: any) {
+      let message = error.msg || 'Minting failed! Please try again!';
+      if (!error.msg) {
+        if (!error.message) {
+          message = 'Transaction Timeout! Please try again.';
+        } else if (error.message.indexOf('0x138')) {
+        } else if (error.message.indexOf('0x137')) {
+          message = `SOLD OUT!`;
+        } else if (error.message.indexOf('0x135')) {
+          message = `Insufficient funds to mint. Please fund your wallet.`;
+        }
+      } else {
+        if (error.code === 311) {
+          message = `SOLD OUT!`;
+          window.location.reload();
+        } else if (error.code === 312) {
+          message = `Minting period hasn't started yet.`;
+        }
+      }
+
+      setAlertState({
+        open: true,
+        message,
+        severity: 'error',
+      });
+    } finally {
+      setIsUserMinting(false);
+    }
+  };
+
+  useEffect(() => {
+    // Refresh the UI when there are confirmed on-chain updates
+    if (props.candyMachineId) {
+      const candyMachineStateSub = props.connection.onAccountChange(
```

Comment:
we had problems dealing with huge numbers of websocket connections negatively affecting our RPC infra in the past. do we need the websocket subscription here? which updates need to be close to realtime?

Reply:
There are complaints that we don't update the CandyMachine state after minting (updating the remaining number of items). The update is finalized in a timely manner. This subscription is listening for those CMv2 state updates.

Reply:
I'll pull this for now. Lets see if this deprecation of FLP helps.